bluetooth: adv_prov: fast_pair: Increase advertising buffer size (#12199)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

* :ref:`peripheral_fast_pair` sample:

>   * Updated by disabling the :kconfig:option:`CONFIG_BT_SETTINGS_CCC_LAZY_LOADING` Kconfig option as a workaround fix for the `Zephyr issue #61033`_.

Reviewer: Should we add an entry for the adv_prov subsystem (about increased buffer configuration)?

Author: Hmm I think we should. I will add it. However, I would rather not add known issue for adv_prov subsystem as it's only matter of allowed range. Are you ok with that?

Reviewer: sure, one known issue seems enough

Reviewer 2: Old range does not allow value of 26 (needed when battery data is included in Fast Pair advertising and 10 Account Keys have been written). Maybe we could add a known issue under the Fast Pair advertising provider instead and link it in the sample's section?

Author: Those are two separate issues. First one is choosing appropriate size for sample. Second one is allowing appropriate range in Advertising Providers. I can add them both but I would rather not link them to not mix it up.

Author: My initial thought was to not add the range issue as I think it's neglectable, but I can add it if you insist.

Reviewer 2: Definitely this should be one known issue, not two. Still, it's an issue of an advertising data provider, not sample itself (currently sample does not even configure this option), so I would put it under the advertising data provider's section (if you think it's needed, we could also link it under sample's section). Then you could merge the fix first and then link commit that needs to be cherry-picked to fix the issue as workaround.

Author: I disagree. It's sample responsibility to set appropriate Kconfig value. Using default value is also a choice.

Reviewer 2: Discussed offline. Agreed to move the known issue under the advertising provider.

>   * Fixed NCSDK-23315 known issue where the sample was unable to advertise in Fast Pair not discoverable advertising mode when it had 5 Account Keys written.

Reviewer 3: ```suggestion
  * Fixed an issue where the sample was unable to advertise in Fast Pair not discoverable advertising mode when it had five Account Keys written.
```

  * Updated by deleting reset in progress flag from settings storage instead of storing it as ``false`` on factory reset operation.
    This is done to ensure that no Fast Pair data is left in the settings storage after the factory reset.

* :ref:`bt_le_adv_prov_readme` library:

>   * Changed the :kconfig:option:`CONFIG_CONFIG_BT_ADV_PROV_FAST_PAIR_ADV_BUF_SIZE` allowed range and increased its default value to 20.
>     This is done to align the buffer size to new, 1 byte larger, Fast Pair not discoverable advertising data size.

Reviewer 3: ```suggestion
  * Changed the allowed range of the :kconfig:option:`CONFIG_CONFIG_BT_ADV_PROV_FAST_PAIR_ADV_BUF_SIZE` Kconfig option and set its default value to 20.
    This is done to increase the buffer size to 1 byte more than the Fast Pair not discoverable advertising data size.
```

Reviewer 3: ah, I understood it wrongly. I have updated the comment. I think that we need not mention "I byte more", since user could make out from the new value set for the Kconfig option.

>     This is done to align the buffer size to new Fast Pair not discoverable advertising data size.

Reviewer 2: `... data size after salt size was increased from 1 byte to 2 bytes.`? (we may be even more explicit here I think
